src/DataCollector/DebugPlugin.php

```diff
@@ -0,0 +1,149 @@
+            ActionEventEmitterEventStore::EVENT_LOAD,
+            ActionEventEmitterEventStore::EVENT_APPEND_TO,
+            ActionEventEmitterEventStore::EVENT_DELETE,
+        ];
+        foreach ($trackedActionEvents as $eventStoreEvent) {
+            $eventStore->attach($eventStoreEvent, function (ActionEvent $event) use ($eventStoreEvent, $eventStore) {
+                if ($event->getName() === $eventStore::EVENT_CREATE) {
+                    $stopName = sprintf('%s:%s', $event->getName(), $event->getParam('stream')->streamName());
+                } else {
+                    $stopName = sprintf('%s:%s', $event->getName(), $event->getParam('streamName'));
+                }
+
+                $this->stopwatch->start($stopName);
+
+            }, 1000);
+
+            $eventStore->attach($eventStoreEvent, function (ActionEvent $event) use ($eventStoreEvent) {
+                if ($event->getName() === 'create') {
+                    $stopName = sprintf('%s:%s', $event->getName(), $event->getParam('stream')->streamName());
+                } else {
+                    $stopName = sprintf('%s:%s', $event->getName(), $event->getParam('streamName'));
+                }
+                $time = $this->stopwatch->stop($stopName);
+
+                $debugEvent = new DebugEvent($event, $time);
+
+                $this->eventStoreActions[] = $debugEvent;
+            }, -1000);
+        }
+    }
+
+    public function detachFromEventStore(ActionEventEmitterEventStore $eventStore): void
+    {
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getName()
+    {
+        return 'event_store';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function collect(Request $request, Response $response, \Exception $exception = null)
+    {
+        $this->data = [];
+        foreach ($this->eventStoreActions as $debugEvent) {
+            // Don't count non-returned streams
+            $debugEvent->getStreamName() === null ?: $allStreams[] = $debugEvent->getStreamName();
+
+            /** @var DebugEvent $debugEvent */
+            $actionEvent = $debugEvent->getEvent();
+            $entry['event_name'] = $actionEvent->getName();
+
+            if($actionEvent->getName() === TransactionalActionEventEmitterEventStore::EVENT_BEGIN_TRANSACTION) {
+                $entry['duration'] = 0;
+            } else {
+                $entry['duration'] = $debugEvent->getStopwatchEvent()->getDuration();
+            }
+
+            $entry['stream_events'] = $debugEvent->getStreamEvents();
+            $entry['stream_name'] = $debugEvent->getStreamName();
+            $this->data['event_store_actions'][] = $entry;
+
+            $this->data['metrics']['total_events']++;
+            $this->data['metrics']['total_messages'] += count($entry['stream_events']);
+            $this->data['metrics']['total_duration'] += $entry['duration'];
+        }
+        $this->data['metrics']['total_streams'] = count(\array_unique($allStreams));
```

Comment:
same as above, you don't need `\` for php functions

Reply:
Benchmark or it ain't happened :laughing: 

More seriously,
PHP function calls are usually not namespaced to easily differentiate them from user functions.
This is a coding standard that is applied on many PHP projects, Symfony included.

Furthermore, don't you think think this is a good example of the famous "premature optimization is the root of all evil" quote?

Thanks for your time.

Reply:
You can find a benchmark here: https://veewee.github.io/blog/optimizing-php-performance-by-fq-function-calls/

Reply:
I agree it is not actually needed. Most packages installed by composer are not using the backslashes anyway. If you want to speed up your application it would be better to use something like https://github.com/dg/composer-backslasher. I didn't try it yet so I can't tell you if it has any downsides.

Reply:
Thanks for the linkgs @enumag, it's always interesting to dive into these benchmarks :)

Although the gain is quite small (it takes 2.26% more time to execute unamespaced function), it's here!